.gitignore

```diff
@@ -15,3 +15,6 @@
 # vendor/
 
 bin/
+
+# Intellij idea directory
+/.idea/
```

Comment:
@ldmi3i Just for curiosity, why are we using intellij here?

Like is that required and if yes what are its advantages?

Reply:
It is general practice to add all editors specific data to gitignore cause they are not related to the project. (to not commit them by mistake)

I don’t know who uses what editors, but I think everyone who will have same specific editor files will also add them to ignore. I just don't know all possible files to ignore of other editors  (I think a lot of people use visual code and other ide/editors and we'll need to add that files too) .

It's just I'm using IntelliJ Idea so added it files to ignore .

Reply:
@ldmi3i there was a similar chat in the template repo regarding `.vscode` being included in the gitignore.

From what I recall, the consensus reached there with @shukra-in-spirit was that we should not include individual editors in this file here - because of the if X then why not Y problem :)

Reply:
Also @ldmi3i, one other thing I forgot to mention.

You or anyone else of us here will have to include the `sonar cloud token` similar to the template repo, the respective check is failing below due to the missing env var for the container.

Reply:
Mm that's very strange cause it's the main purpose of this ignore file to protect us from commiting wrong files. Wherever I worked these files were added there. Usually gitignore is created immediately with all possible directories for all editors (not so much editors which create such files).

Reply:
> Also @ldmi3i, one other thing I forgot to mention.
> 
> You or anyone else of us here will have to include the `sonar cloud token` similar to the template repo, the respective check is failing below due to the missing env var for the container.

Not really, here problem not with token. Here it can't find pull request (don't understand how can it be)

Reply:
Wow, error updates to new
ERROR: Error during SonarScanner execution
[52](https://github.com/rog-golang-buddies/api-hub_data-scraping-service/runs/7371745269?check_suite_focus=true#step:4:53)
ERROR: Could not find a default branch to fall back on.

Maybe we need to add service name or something like that to Sonar?

Reply:
Also there is PR s where that check passed
https://github.com/rog-golang-buddies/api-hub_data-scraping-service/pull/8

Reply:
Maybe, I’m not that familiar with the behavior of this workflow 📦

Reply:
@ldmi3i on the gitignore front, that’s totally your call.
I don’t see an issue with it - just that for example I can say I’m using editor X why are you not including that in the gitignore here 🗡️ 

But yea that’s not really important atm.

Reply:
Yea, I see the passing PR. Strange this, since it works fine in template repo heh

Reply:
@ldmi3i maybe the Sonar cloud key might be misconfigured due to some reason? Like is it referring to the key you’ve provided?

just a guess, might be wrong

Reply:
I'm not including other editors cause I'm lazy  and just don't want to search full gitignore file and waste time cause it may do everyone who face this problem)
Okay, I'll update gitignore with other editors.

Reply:
> @ldmi3i maybe the Sonar cloud key might be misconfigured due to some reason? Like is it referring to the key you’ve provided?
> 
> just a guess, might be wrong

Yes, probably key here is the reason. But before this I retrieved next error:
https://github.com/rog-golang-buddies/api-hub_data-scraping-service/runs/7370823731?check_suite_focus=true

<details><summary>Log</summary>
ERROR: Error during SonarScanner execution
[53](https://github.com/rog-golang-buddies/api-hub_data-scraping-service/runs/7370823731?check_suite_focus=true#step:4:54)
ERROR: Could not find the pullrequest with key '7'
[54](https://github.com/rog-golang-buddies/api-hub_data-scraping-service/runs/7370823731?check_suite_focus=true#step:4:55)
ERROR: Caused by: Error 404 on https://sonarcloud.io/api/alm_integration/show_pullrequest?project=rog-golang-buddies_golang-template-repository&pullrequestKey=7 : {"errors":[{"msg":"The pullrequest could not be found"}]}
</details>

Reply:
And I have read that this project key must be unique per project
https://stackoverflow.com/questions/37749665/what-are-standards-for-sonar-projectkey-sonar-projectname-sonar-projectversio
And I was wrong - the project key should be just api-hub

Reply:
Ohk

Reply:
It’s fine if you leave out other editors though. Me personally I always make sure not to include those files in a commit 📟

Reply:
I and my coworkers tried avoiding commit such files (it was some docker/ci local changes for testing), but it finally ended with pushing this changes by mistake. Once we restarted prod because of such mistake (ci changes to deploy prod from dev branch passed to master ...)
I think this is a good topic for a discussion channel)

Reply:
> And I was wrong - the project key should be just api-hub

May be not wrong) currently don't understand this projectKey . Can it be one for several repos or one repo = one project for sonar. Will read about it.

Reply:
@shukra-in-spirit knows more about this in case you don’t find anything specific 👍

Reply:
Okay thank you.
And I've done some investigation about gitignore. I didn't know about  global gitignore, applied it locally and now editor files ignored and I think it's the best solution. So I'm restoring gitignore. Thanks for your remark 👍

Reply:
No worries